src/cart/src/cart/crt_group.c

```diff
@@ -2506,8 +2512,9 @@ grp_add_to_membs_list(struct crt_grp_priv *grp_priv, d_rank_t rank)
 	}
 
 	/* Regenerate linear list*/
-	grp_regen_linear_list(grp_priv);
-
+	ret = grp_regen_linear_list(grp_priv);
+	if (ret != 0 && rc == 0)
```

Comment:
on ret != 0 error this needs then handling of:
grp_add_free_idx(&grp_priv->gp_membs.cgm_free_indices, index, false);
membs->rl_ranks[index] = CRT_NO_RANK;
grp_priv->gp_size--;

Reply:
What if grp_add_free_idx() fails?

Reply:
then we are screwed:)

Reply:
in reality it would mean we would have a hole in membs array that cannot be used anymore by anyone, since membs[index] would be useable, but that index will not appear in 'free indecies' list